### ISSUE TITLE
Add unit tests

### ISSUE THREAD

**Issue by andrewda** [SouthEugeneRoboticsTeam/sertain#14]
![Test](https://raw.githubusercontent.com/swiftingio/blog/%2317-Unit-test-all-the-things/unit-test-all-the-things.png?utm_source=swifting.io&utm_medium=web&utm_campaign=blog%20post)

Unit tests are crucial to ensure that weird math stuff (e.g. trajectories and metric conversions) don't break with a new change. It'd also be good to eventually implement some integration and e2e tests to ensure it works in a sample robot simulation.

**Comment by gnawinganimal** [comment]
Yeah... 😬 its kind of important. Fortunately kotlin has a really nice testing system that requires almost no setup.
